controllers/checlusterrestore/backup_data_restorer.go

```diff
@@ -209,6 +225,19 @@ func cleanPreviousInstallation(rctx *RestoreContext, dataDir string) (bool, erro
 	return true, nil
 }
 
+func waitPreviousInstallationDeleted(rctx *RestoreContext) (bool, error) {
```

Comment:
Should we wait some time and than cleanup finalizers by ourselves
 ?

Reply:
I think it should be done in the main controller. I'll provide some refactoring in a separate PR.